services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/taydennys/TaydennysControllerITest.kt

```diff
@@ -571,4 +575,54 @@ class TaydennysControllerITest(@Autowired override val mockMvc: MockMvc) : Contr
             }
         }
     }
+
+    @Nested
+    inner class Delete {
+        private val url = "/taydennykset/$id"
+
+        @Test
+        @WithAnonymousUser
+        fun `returns 401 when user is unknown`() {
+            delete(url).andExpect(status().isUnauthorized).andExpect(hankeError(HankeError.HAI0001))
+
+            verify { taydennysService wasNot Called }
+        }
+
+        @Test
+        fun `returns 404 when no taydennys`() {
+            every {
+                taydennysAuthorizer.authorize(id, PermissionCode.EDIT_APPLICATIONS.name)
+            } throws TaydennysNotFoundException(id)
+
+            delete(url).andExpect(status().isNotFound).andExpect(hankeError(HankeError.HAI6001))
+
+            verify { taydennysAuthorizer.authorize(id, PermissionCode.EDIT_APPLICATIONS.name) }
+        }
+
+        @Test
+        fun `returns 404 when no application`() {
+            every {
+                taydennysAuthorizer.authorize(id, PermissionCode.EDIT_APPLICATIONS.name)
+            } throws HakemusNotFoundException(1L)
+
+            delete(url).andExpect(status().isNotFound).andExpect(hankeError(HankeError.HAI2001))
+
+            verify { taydennysAuthorizer.authorize(id, PermissionCode.EDIT_APPLICATIONS.name) }
+        }
+
+        @Test
+        fun `deletes taydennys`() {
+            every {
+                taydennysAuthorizer.authorize(id, PermissionCode.EDIT_APPLICATIONS.name)
+            } returns true
+            every { taydennysService.delete(id, USERNAME) } just Runs
```

Comment:
Everywhere else, this has been expressed like:
```suggestion
            justRun { taydennysService.delete(id, USERNAME) }
```

Reply:
Thanks, changed it.